docs/proposals/metrics-api.md

```diff
@@ -0,0 +1,134 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/metrics-api.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Resource Metrics API
+
+*This proposal is based on and supersedes [compute-resource-metrics-api.md](compute-resource-metrics-api.md).*
+
+This document describes API part of MVP version of Resource Metrics API effort in Kubernetes.
+Once the agreement will be made the document will be extended to also cover implementation details.
+This document may be also a subject of changes once we will have more well-defined use cases.
+
+## Goal
+
+The goal for the effort is to provide resource usage metrics for pods and nodes through the API server.
+This will be a stable, versioned API which core Kubernetes components can rely on.
+In the first version only the well-defined use cases should be handled,
+altough the API should be easily extensible for potential future use cases.
+
+## Use cases
+
+This paragraph describes well-defined use cases which should be handled in the first version.
+Use cases which are not listed below are out of the scope of MVP version of Resource Metrics API.
+
+#### Horizontal Pod Autoscaler
+
+HPA needs the latest value of cpu usage (as an mean from some time window)
+for the given set of pods (both pod list or label selector are fine for this purpose).
+
+#### Scheduler
+
+Scheduler in order to schedule best-effort pods requires node level resource usage metrics.
+
+#### kubectl top
+
+This feature is not yet specified/implemented although it seems reasonable to support the following options:
+- ```kubectl top node <id>``` - list all pods running on the given node accompanied with theirs cpu/mem usage
+- ```kubectl top pod <id>``` - print cpu usage stats for the given pod
+- ```kubectl top pod --selector=<label-query>``` - list all pods defined by the label query accompanied with theirs cpu/mem usage
+
+#### Kubernetes dashboard
+
+[Kubernetes dashboard](https://github.com/kubernetes/dashboard) in order to draw graps requires resource usage
+in timeseries format form the last 15 minutes. Alternatively just the latest usage value is fine.
+
+## Proposed API
+
+The metrics API will be in a separate [API group](api-group.md) called ```metrics```.
+
+#### Schema
+
+The proposed schema is as follow. Each object has `TypeMeta` and `ObjectMeta` fields
+to be compatible with Kubernetes API standards.
+
+```go
```

Comment:
We have an API convention that if an API Group is present, then all types of the api group should be present.

We also think we may want, in the future, to have clusters whose set of APIs does not expose Nodes, and thus does not expose metrics about nodes. 

Therefore, the Node and Pod metrics should be in different API groups. 

So, we should structure this as something like:

```go
// File: pkg/apis/inframetrics/v1beta1/types.go:
type Node struct {
 TypeMeta
  ObjectMeta
  Metrics Metrics
 }
type Metrics struct {
...
}

// File: pkg/apis/containermetrics/v1beta1/types.go:
type Pod struct {
 TypeMeta
  ObjectMeta
  Metrics Metrics
 }
type Metrics struct {
...
}
```

You make that transition from 1 api group to 2 api groups either when you move from `v1alpha1` to `v1beta1`, or when you move from `v1beta1` to `v1`.   But, I'd like to split them before you go to GA.